app/services/create_order_service.rb

```
@@ -26,13 +26,21 @@ def self.artwork_price(external_artwork, edition_set_id: nil)
     if edition_set_id
       edition_set = find_edition_set(external_artwork, edition_set_id)
       raise Errors::OrderError, 'Unknown edition set.' unless edition_set
-      edition_set[:price]
+      price_in_cents(edition_set[:price_listed], external_artwork[:currency])
     else
-      external_artwork[:price]
+      price_in_cents(external_artwork[:price_listed], external_artwork[:currency])
     end
   end
 
   def self.find_edition_set(external_artwork, edition_set_id)
     external_artwork[:edition_sets].find { |e| e[:id] == edition_set_id }
   end
+
+  # TODO: 🚨 update gravity to expose amount in cents and remove this duplicate logic
+  # https://github.com/artsy/gravity/blob/65e398e3648d61175e7a8f4403a2d379b5aa2107/app/models/util/for_sale.rb#L221
+  def self.price_in_cents(price_in_dollars, currency)
+    raise Errors::OrderError 'No price found' unless price_in_dollars&.positive?
+    raise Errors::OrderError 'Invalid currency' if currency.present? && currency != 'USD' # TODO: handle other currencies
```

Do we also want to raise errors when currency is not present?